Assets/Scripts/UI/UIManager.cs

```diff
@@ -0,0 +1,52 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+using TMPro;
+public class UIManager : MonoBehaviour
+{
+    [SerializeField] private Slider uiCombo, uiHealth;
+    [SerializeField] private TMP_Text hasTnt, uiLoop;
+    [SerializeField] private Image uiTnt;
+
+    bool oldHasTnt;
+
+    //POINTERS
+    ComboBehaviour thisCombo;
+    CharacterBehaviour thisPlayer;
+
+    private void Awake()
+    {
+        //ASSIGN POINTERS
+        thisCombo = FindObjectOfType<ComboBehaviour>();
+        thisPlayer = FindObjectOfType<CharacterBehaviour>();
+
+        //POINTER PROPERTIES
+        uiCombo.maxValue = thisCombo.maxCombo;
+        uiCombo.value = 0;
+    }
+
+    void Update()
+    {
+        uiCombo.value = thisCombo.currentCombo;
+        hasTnt.text = "hasTnt: " + CarriageManager.playerHasTnt;
+
+    }
+    public void UpdateTntUi(bool _hasTnt)
+    {
+        switch (_hasTnt)
```

Comment:
little point `switch`ing on a boolean. just use `if`

```
if (_hasTnt){
    Color _color = uiTnt.color;
    _color.a = 1;
    uiTnt.color = _color;
} else {
    Color _color2 = uiTnt.color;
    _color2.a = 0;
    uiTnt.color = _color2;
}
```

Reply:
Will keep note. But kept as is for now.